### ISSUE TITLE
spack-stack release 1.5.1

### ISSUE THREAD

**Issue by climbfuji** [JCSDA/spack-stack#819]
**Is your feature request related to a problem? Please describe.**
We are planning a spack-stack release towards the end of the month (October 2023), which will be rolled out over a few days into early November. A project board was created for the release to capture all the PRs that go into there. So far, we've been trying to merge all changes into develop and release/1.5.1 in parallel (cherry-picking), but we should do a final merge of the release/1.5.1 branches (which were created from the release/1.5.0 branches) back into develop after the release.

Below is the checklist:

- [x] Finalize code updates for release 1.5.1, except any potential site config updates and documentation updates
    - Project board: https://github.com/orgs/JCSDA/projects/4
- [x] Roll out release, potential site config updates go back into the release branch
    - [x] Hercules (@climbfuji): `/work/noaa/epic/role-epic/spack-stack/hercules/spack-stack-1.5.1/envs/unified-env`
        - [x] https://github.com/JCSDA/spack-stack/pull/849
    - [x] Orion (EPIC) `/work/noaa/epic/role-epic/spack-stack/orion/spack-stack-1.5.1` (@ulmononian)
    - [x] Discover (@climbfuji) `/gpfsm/dswdev/jcsda/spack-stack/spack-stack-1.5.1/envs/unified-env`
        - [x] https://github.com/JCSDA/spack-stack/pull/853
    - [x] Casper (@climbfuji) `/glade/work/epicufsrt/contrib/spack-stack/casper/spack-stack-1.5.1/envs/unified-env`
        - [x] https://github.com/JCSDA/spack-stack/pull/855
    - [x] ~~Cheyenne (EPIC)~~ last supported release on Cheyenne is 1.5.0
    - [x] Derecho (~~EPIC~~ JCSDA this time because of https://github.com/JCSDA/spack-stack/issues/835)
        - [x] https://github.com/JCSDA/spack-stack/pull/853
    - [x] Acorn (EMC) @AlexanderRichert-NOAA
        - [x] https://github.com/JCSDA/spack-stack/pull/853
    - [x] Gaea C4 (@ulmononian) `/lustre/f2/dev/wpo/role.epic/contrib/spack-stack/c4/spack-stack-1.5.1/envs/unified-env`
        - [x] https://github.com/JCSDA/spack-stack/pull/855
    - [x] Gaea C5 (@climbfuji) `/lustre/f2/dev/wpo/role.epic/contrib/spack-stack/c5/spack-stack-1.5.1/envs/unified-env`
        - [x] https://github.com/JCSDA/spack-stack/pull/853
    - [x] Hera (@RatkoVasic-NOAA) `/scratch1/NCEPDEV/nems/role.epic/spack-stack/spack-stack-1.5.1`
        - [x] https://github.com/JCSDA/spack-stack/pull/855
    - [x] Jet (@RatkoVasic-NOAA) `/mnt/lfs4/HFIP/hfv3gfs/role.epic/spack-stack/spack-stack-1.5.1`
        - [x] https://github.com/JCSDA/spack-stack/pull/855
    - [x] Narwhal (@climbfuji)
        - [x] https://github.com/JCSDA/spack-stack/pull/853
    - [x] Nautilus (@climbfuji)
        - [x] https://github.com/JCSDA/spack-stack/pull/853
    - [x] S4 (@climbfuji) `/data/prod/jedi/spack-stack/spack-stack-1.5.1/envs/unified-env`
        - [x] https://github.com/JCSDA/spack-stack/pull/853
    - [x] Parallel Works (JCSDA installations; @climbfuji) `/contrib/spack-stack/spack-stack-1.5.1/envs/unified-env`
        - [x] AWS
        - [x] Azure
        - [x] Gcloud
        - [x] https://github.com/JCSDA/spack-stack/pull/855
    - [ ] Parallel Works (EPIC installations; @natalie-perlin)
        - [ ] AWS
        - [ ] Azure
        - [ ] Gcloud
    - [x] AWS ParallelCluster (@climbfuji): `/mnt/experiments-efs/skylab-v7/spack-stack-1.5.1/envs/unified-env`
        - [x] https://github.com/JCSDA/spack-stack/pull/853
    - [x] AWS Single-node AMI (@climbfuji) **in progress**
        - [x] https://github.com/JCSDA/spack-stack/pull/855
    - [x] JEDI CI containers (@climbfuji )
        - [x] intel-impi: uploaded with `test` tag to AWS ECR JCSDA-NOAA us-east-1 / JCSDA-USAF us-east-2
        - [x] gnu-openmpi: uploaded with `test` tag to AWS ECR JCSDA-NOAA us-east-1 / JCSDA-USAF us-east-2
        - [x] clang-mpich: uploaded with `test` tag to AWS ECR JCSDA-NOAA us-east-1 / JCSDA-USAF us-east-2
        - [x] https://github.com/JCSDA/spack-stack/pull/855
- [x] Final documentation updates
- [x] Tag spack, revert special changes in spack-stack (`.gitmodules` and container recipes), tag spack-stack
- [x] Create release notes and publish release
- [x] Merge release branches back into develop/jcsda_emc_spack_stack: created follow-up issue to track this #861

**Comment by climbfuji** [comment]
Completed
